openmdao/solvers/solver.py

```diff
@@ -616,6 +617,19 @@ def _solve(self):
                         stall_count += 1
                         if stall_count >= stall_limit:
                             stalled = True
+                        if stall_count == 3 and not self.linesearch.options['print_bound_enforce']:
```

Comment:
One more thing. I don't think we want to run `_single_iteration` twice in the same iteration, because it may be inefficient if the submodel contains solvers or slow components.  I think it would be better to move this new block to the point at the beginning of the recorder context where we already run _single_iteration.  If the conditional is met, we warn, set the option, and run single iteration, otherwise, we just run the single iteration.